src/main/kotlin/org/rust/ide/core/overrideImplement/generateUtil.kt

```diff
@@ -0,0 +1,103 @@
+package org.rust.ide.core.overrideImplement
+
+import com.intellij.codeInsight.generation.ClassMember
+import com.intellij.codeInsight.generation.MemberChooserObject
+import com.intellij.codeInsight.generation.MemberChooserObjectBase
+import com.intellij.ide.util.MemberChooser
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.ui.SimpleColoredComponent
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.mixin.*
+import org.rust.lang.core.psi.util.trait
+import org.rust.utils.getFormattedParts
+import javax.swing.JTree
+
+sealed class TraitMember {
+    class Function(val function: RsFunction) : TraitMember()
+    class Type(val type: RsTypeAlias) : TraitMember()
+    class Constant(val constant: RsConstant) : TraitMember()
+}
```

Comment:
I decided to use sealed class to make the code cleaner and make all `when`s exhaustive, but now it is not so important.